### PR TITLE
fix: toast to snackbar for login work

### LINKED ISSUE #797: fix login fragment UI
**Screenshots**

screenshot_2018-12-30-12-37-04-235_com eventyay attendee
screenshot_2018-12-30-12-25-53-760_com eventyay attendee

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by GOVINDDIXIT**
Part for  #785 
fix #797
merge after #790

Screenshots for Login fragment UI fix
screenshot_2018-12-30-13-19-27-496_com eventyay attendee
screenshot_2018-12-30-13-19-38-194_com eventyay attendee

GIF for the Snackbar change:
ezgif com-video-to-gif 2

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 1]
```diff
@@ -67,11 +67,13 @@ class LoginFragment : Fragment() {
...
 
         loginViewModel.error.observe(this, Observer {
-            Toast.makeText(context, it, Toast.LENGTH_LONG).show()
+            Snackbar.make(getActivity()!!.findViewById(android.R.id.content),
+                it, Snackbar.LENGTH_LONG).show()
         })
 
         loginViewModel.loggedIn.observe(this, Observer {
-            Toast.makeText(context, getString(R.string.welcome_back), Toast.LENGTH_LONG).show()
+            Snackbar.make(getActivity()!!.findViewById(android.R.id.content),
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: use safe calls instead of `!!`

Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: but then it is showing type mismatch as snackbar requires view of type View and we are giving View? Pls guide

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: First of all, don't use findViewById. Secondly, that'll fix it

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: `android.R.id.content` This isn't to be used. Use coordinator layout of the current fragment/activity

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Yeah, OK. Use CoordinatorLayout of the root element

**Comment by iamareebjamal** [comment]
Your PR has undone several changes

**Comment by GOVINDDIXIT** [comment]
Yes. Sorry for that I will correct them using different branch

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt (hunk 1 above)
Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: but how to get CoordinatorLayout of the root element. when I try to use CoordinatorLayout of root element i.e rootLayout(activity_main.xml) It is becoming null
I'm facing
`java.lang.IllegalStateException: rootView.findViewById(R.id.rootLayout) must not be null
        at org.fossasia.openevent.general.order.OrdersUnderUserFragment.onCreateView(OrdersUnderUserFragment.kt:48)`
Need help

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Call findViewById on activity

**Comment by iamareebjamal** [comment]
OK

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 2]
```diff
@@ -56,7 +58,7 @@ class LoginFragment : Fragment() {
         savedInstanceState: Bundle?
     ): View? {
         rootView = inflater.inflate(R.layout.fragment_login, container, false)
-
+        CoordinatorLayout = rootView.findViewById(R.id.loginCoordinatorLayout)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Don't use findViewById

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: I have no idea why is this capital. Is this a class?

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt [hunk 3]
```diff
@@ -48,23 +50,24 @@ class ProfileFragment : Fragment() {
...
+        CoordinatorLayout = rootView.findViewById(R.id.profileCoordinatorLayout)
 
         setHasOptionsMenu(true)
 
+        if (!profileViewModel.isLoggedIn()) {
+            Snackbar.make(
+                CoordinatorLayout, "You need to log in first!", Snackbar.LENGTH_SHORT).show()
+            Handler().postDelayed({
+                redirectToLogin()
+            }, 1000)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: Strange indentation

**Comment by GOVINDDIXIT** [comment]
@iamareebjamal  I think the current way to show snackbar as shown in GIF is more apt as showing snackbar from bottom of screen will hinder the UI nor it is properly hiding it. It is hiding navigation buttons upto 3/4. If you agree with current config I will move toward for requested changes

File: app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt (hunk 3 above)
Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/ProfileFragment.kt]: will correct don't know why autoformat is going for this

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt (hunk 2 above)
Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: > Don't use findViewById

Pls suggest alternate way

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 4]
```diff
@@ -36,6 +38,7 @@ class LoginFragment : Fragment() {
 
     private val loginViewModel by viewModel<LoginViewModel>()
     private lateinit var rootView: View
+    private lateinit var coordinatorLayout: CoordinatorLayout
```
[the remark this answers is not in the document]
Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: > Step 'ktlint' found problem in 'app\src\main\java\org\fossasia\openevent\general\auth\LoginFragment.kt':
Error on line: 90, column: 1
Exceeded max line length (120)
@iamareebjamal  can we increase this 120 word limit

[the remark this answers is not in the document]
Comment (GOVINDDIXIT) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: on directly using rootview.requiredCoordinatorLayout. I am getting above error

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Then break that line

**Pull request merged**